src/dodal/plans/scanspec.py

```diff
@@ -0,0 +1,65 @@
+import operator
+from functools import reduce
+from typing import Annotated, Any
+
+import bluesky.plans as bp
+from bluesky.protocols import Movable, Readable
+from cycler import Cycler, cycler
+from pydantic import Field, validate_call
+from scanspec.specs import Spec
+
+from dodal.common import MsgGenerator
+from dodal.plan_stubs.data_session import attach_data_session_metadata_decorator
+
+
+@attach_data_session_metadata_decorator()
+@validate_call(config={"arbitrary_types_allowed": True})
+def spec_scan(
```

Comment:
FYI: I added https://github.com/DiamondLightSource/mx-bluesky/issues/596 as I think we have overlap here

Reply:
This plan has now moved repositories so many times that the issue has now been lost to time, but it does need improvement. The `axes_to_move` parameter was a temporary fix and we should try to get rid of it.

Reply:
https://github.com/DiamondLightSource/dodal/issues/866

Reply:
I think we no longer need it, I just need to find time to start up blueapi with these plans and ensure we can run a spec just passing the name of the device and no axes_to_move

Reply:
In theory, all plans that use a ScanSpec should be constructing their specific spec and passing it to this for software scans) or an ophyd plan to-be-produced for hardware.